test/parallel/council-pallet.API.foundation.maintenance.test.ts

```diff
@@ -0,0 +1,245 @@
+      );
+      const propAfter = await getProposal(hash);
+      validate(events, propAfter, propBefore);
+    },
+  );
+  it("Test that sudo address can veto an already voted proposal", async () => {
+    const { validate } = testCases["Foundation"];
+    const hash = proposalHashes[5];
+    const propBefore = await getProposal(hash);
+    await voteProposal(hash, councilUsers);
+    const events = await Sudo.asSudoFinalized(Sudo.sudo(Council.veto(hash)));
+    const propAfter = await getProposal(hash);
+    validate(events, propAfter, propBefore);
+  });
+  it("Test that sudo address cannot close an already voted proposal", async () => {
+    const validate = validateNoOK;
+    const hash = proposalHashes[8];
+    const propBefore = await getProposal(hash);
+    await voteProposal(hash, councilUsers);
+    const propIndex = JSON.parse(JSON.stringify(await getVotes(hash))).index;
+    const events = await Sudo.asSudoFinalized(
+      Sudo.sudo(Council.close(hash, propIndex)),
+    );
+    const propAfter = await getProposal(hash);
+    validate(events, propAfter, propBefore);
+  });
+});
+
+function validateOK(
+  events: MangataGenericEvent[],
+  propAfter: Option<Call>,
+  propBefore: Option<Call>,
+) {
+  expectMGAExtrinsicSuDidSuccess(events);
+  expect(propAfter.toHuman()).toBeNull();
+  expect(propBefore.toHuman()).not.toBeNull();
+}
+function validateNoOK(
+  events: MangataGenericEvent[],
+  propAfter: Option<Call>,
+  propBefore: Option<Call>,
+) {
+  expectMGAExtrinsicSuDidFailed(events);
+  expect(propAfter.toHuman()).not.toBeNull();
+  expect(propBefore.toHuman()).not.toBeNull();
+}
+async function createProposals(users: User[], num = 10): Promise<string[]> {
+  const userToSubmit = users[0];
+  const txs: Extrinsic[] = [];
+  for (let i = 0; i < num; i++) {
+    const tx = Council.propose(
+      users.length,
+      api.tx.sudoOrigin.sudo(
+        api.tx.tokens.mint(
+          MGA_ASSET_ID,
+          userToSubmit.keyRingPair.address,
+          new BN(i).addn(1),
+        ),
+      ),
+      44,
+    );
+    txs.push(tx);
+  }
+  const events = await Sudo.asSudoFinalized(
+    Sudo.sudoAs(userToSubmit, Sudo.batch(...txs)),
+  );
+  return events
+    .filter((x) => x.event.method === "Proposed")
+    .flatMap((x) => x.eventData[2].data.toString());
+}
+async function getProposal(hash: string) {
+  return await api.query.council.proposalOf(hash);
+}
+async function getVotes(hash: string) {
+  return await api.query.council.voting(hash);
+}
+async function voteProposal(hash: string, councilUsers: User[]) {
+  const txs: Extrinsic[] = [];
+  const proposal = await getVotes(hash);
+  for (let i = 0; i < councilUsers.length; i++) {
+    const tx = Sudo.sudoAs(
+      councilUsers[i],
+      Council.vote(
+        hash,
+        JSON.parse(JSON.stringify(proposal.toHuman())).index,
+        "aye",
+      ),
+    );
+    txs.push(tx);
+  }
+  return await Sudo.batchAsSudoFinalized(...txs);
```

Comment:
I guess that those aux functions are duplicated on both tests,  could you please reuse them?

Reply:
Done